File: CustomView_Canvas/app/src/main/java/wang/ralf/customview_canvas/MainActivity.java
package wang.ralf.customview_canvas;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

/**
 * @author Ralf
 */
public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById(R.id.clip_btn).setOnClickListener(this);
        findViewById(R.id.save_btn).setOnClickListener(this);
    }

    @Override
    public void onClick(View v) {
        int id = v.getId();
        Intent intent = null;
        if (id == R.id.clip_btn) {
            intent = new Intent(MainActivity.this, ClipActivity.class);
        } else if (id == R.id.save_btn) {
            intent = new Intent(MainActivity.this, SaveAndRestoreActivity.class);
        }
        if (intent != null) {
            startActivity(intent);
        }
    }
}

File: ViewEventTest1/app/src/main/java/com/ralf/vieweventtest1/MainActivity.kt
package com.ralf.vieweventtest1

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        single_view.setOnClickListener {
            startActivity(Intent(this, SingleViewActivity::class.java))
        }

        single_view_group.setOnClickListener {
            startActivity(Intent(this, SingleViewGroupActivity::class.java))
        }

        viewgroup_view.setOnClickListener {
            startActivity(Intent(this, ViewGroupAndViewActivity::class.java))
        }

        viewgroup_viewgroup.setOnClickListener {
            startActivity(Intent(this, ViewGroupAndVGActivity::class.java))
        }
        viewgroup_viewgroup_view.setOnClickListener {
            startActivity(Intent(this, EventTransmitActivity::class.java))
        }
    }
}
